src/FileNavigator.java

```
@@ -0,0 +1,61 @@
+import java.util.*;
+
+public class FileNavigator {
+    private Map<String, List<FileData>> files;
+
+    public FileNavigator() {
+        this.files = new HashMap<>();
+    }
+
+    public void add(FileData fileData) {
+        if (!fileData.getPath().equals(getPathKey(fileData.getPath()))) {
+            System.out.println("Error: Path-key and file path do not match");
+            return;
```

You should throw an exception here